### PR TITLE
✨Add User Rules Page

### PR DESCRIPTION
**Pull request by Aibono1225**
Relates to https://github.com/SSWConsulting/SSW.Rules/issues/1046

Changes:
- Added a user rules page to display a user's rule data fetched from the Github API
- Added pagination at the bottom of the list of rules

image

**Figure: the new page**

### CODE REVIEW

**Review by bradystroud — CHANGES_REQUESTED** [review]
Looking good 😊

File: src/pages/user-rules.js [hunk 1]
```diff
@@ -0,0 +1,254 @@
...
+                isAscending={isAscending}
+                setIsAscending={setIsAscending}
+              />
+            </div>
+            <div className="text-center mb-4">
+              <button
+                className={`m-3 mx-6 p-2 w-24 rounded-md ${
+                  hasPrevious
+                    ? 'bg-ssw-red text-white'
+                    : 'bg-ssw-grey text-text-gray-400'
```
[the remark this answers is not in the document]
Comment (Aibono1225) [reply src/pages/user-rules.js]: oops it's a mistake 😣

File: src/pages/user-rules.js [hunk 2]
```diff
@@ -0,0 +1,254 @@
...
+        setNotFound(true);
+      }
+    };
+
+    fetchPageData();
+  }, []);
+
+  const fetchGithubData = async (action) => {
+    const githubOwner = 'SSWConsulting';
+    const githubRepo = 'SSW.Rules.Content';
```
[the remark this answers is not in the document]
Comment (Aibono1225) [reply src/pages/user-rules.js]: We didn't have these variables in .env.template, I've just added them🙂

File: src/pages/user-rules.js [hunk 3]
```diff
@@ -0,0 +1,254 @@
...
+    setNextPageCursor(endCursor);
+    setTempCursor(startCursor);
+    setHasPrevious(hasPreviousPage);
+    setHasNext(hasNextPage);
+
+    return data.data?.search;
+  };
+
+  const updateAndFilterRules = (resultList) => {
+    const newResNodesArr = [];
```
[the remark this answers is not in the document]
Comment (Aibono1225) [reply src/pages/user-rules.js]: This is for getting the unique rule title and updating the rule list. The function is too complex, so I've split the logic into smaller functions in my new commit. 🙂

**Comment by Aibono1225** [comment]
image
**Figure: Updated button style**

File: src/pages/user-rules.js [hunk 4]
```diff
@@ -0,0 +1,270 @@
...
+    } else if (action === ActionTypes.AFTER) {
+      setPreviousPageCursor([...previousPageCursor, tempCursor]);
+      cursorQuery = `after: "${nextPageCursor}"`;
+    }
+
+    const query = `
+        {
+          search( query: "repo:${githubOwner}/${githubRepo} is:pr base:${process.env.CONTENT_BRANCH} is:merged sort:updated-desc author:${queryStringRulesAuthor}"
+          type: ISSUE
+          first: 30
```
Comment (bradystroud) [inline src/pages/user-rules.js:81]: Hardcoded values in queries and URLs (e.g., first: 30, first: 20, and https://api.github.com/graphql) should be moved to constants

**Review by bradystroud — CHANGES_REQUESTED** [review]

**Review by bradystroud — APPROVED** [review]

**Pull request merged**